AK/QuickSort.h

```diff
@@ -1,23 +1,38 @@
 /*
  * Copyright (c) 2018-2020, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2022, Marc Luqué <marc.luque@outlook.com>
  *
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
 #pragma once
 
+#include <AK/InsertionSort.h>
 #include <AK/StdLibExtras.h>
 
 namespace AK {
 
-/* This is a dual pivot quick sort. It is quite a bit faster than the single
- * pivot quick_sort below. The other quick_sort below should only be used when
- * you are stuck with simple iterators to a container and you don't have access
- * to the container itself.
- */
+// This is a dual pivot quick sort. It is quite a bit faster than the single
+// pivot quick_sort below. The other quick_sort below should only be used when
+// you are stuck with simple iterators to a container and you don't have access
+// to the container itself.
+//
+// We use a cutoff to insertion sort for partitions of size 7 or smaller.
+// The idea is to avoid recursion for small partitions.
+// The value 7 here is a magic number. According to princeton's CS algorithm class
+// a value between 5 and 15 should work well in most situations:
+// https://algs4.cs.princeton.edu/23quicksort/
+
+static constexpr int INSERTION_SORT_CUTOFF = 7;
```

Comment:
Should the type of `INSERTION_SORT_CUTOFF` not be `int`, maybe `ssize_t` instead?

Reply:
The type that it is compared to is int, see below. This can be changed when the quicksort API changes.